pkg/util/cluster/cluster.go

```diff
@@ -738,10 +738,12 @@ func (c *Cluster) Delete(ctx context.Context, vnetResourceGroup, clusterName str
 			c.log.Errorf("CI E2E cluster %s not found in resource group %s", clusterName, vnetResourceGroup)
 			errs = append(errs, err)
 		}
-		if oc.Properties.ServicePrincipalProfile != nil {
-			errs = append(errs,
-				c.deleteApplication(ctx, oc.Properties.ServicePrincipalProfile.ClientID),
-			)
+		if oc != nil {
```

Comment:
question: Would it make sense to short-circuit and return an error if `oc` is indeed `nil`? 

What should happen if `oc == nil`?